File: 234-palindrome-linked-list/palindrome-linked-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {boolean}
 */
var isPalindrome = function(head) { 
    let slow = head;
    let fast = head;
    while(fast && fast.next){
        slow = slow.next;
        fast = fast.next.next;

    }
    let prev = null;
    let current = slow;
    while(current!=null){
        let temp = current.next; 
        current.next = prev;
          prev = current;
        current = temp;
      
    } 
    let first = head;
    let second = prev;
    while(second){
        if(first.val!=second.val){
            return false;
        }
        first = first.next;
        second = second.next;
    }
    return true;


   

    
};

File: 234-palindrome-linked-list/palindrome-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public boolean isPalindrome(ListNode head) {
       //finding middlevalue 
       ListNode slow = head;
       ListNode fast = head;
       while(fast!=null && fast.next!=null){
        slow = slow.next;
        fast =fast.next.next;

       }
       ListNode current= slow;
      //reverse the secondhalf list 
      ListNode prev = null;
      while(current!=null){
        ListNode temp = current.next;
        current.next = prev;
        prev = current;
        current = temp;

      }
    ListNode first = head;
    ListNode second = prev;
    while(second!=null){
        if(first.val!=second.val){
            return false;
        }
        first = first.next;
        second=second.next;
    }
    return true;
    }
}